frontend/catalyst/device/qjit_device.py

```diff
@@ -443,50 +443,29 @@ def execute(self, circuits, execution_config):
 AnyQJITDevice = Union[QJITDevice, QJITDeviceNewAPI]
 
 
-def filter_out_adjoint(operations):
-    """Remove Adjoint from operations.
+def filter_out_modifiers(operations):
+    """Remove Adjoint/Control from operations.
 
     Args:
-        operations (List[Str]): List of strings with names of supported operations
+        operations (Iterable[str]): list of operation names
 
     Returns:
-        List: A list of strings with names of supported operations with Adjoint and C gates
-        removed.
+        Set: filtered set of operation names
     """
+    control = re.compile(r"^C\(.*\)$")
     adjoint = re.compile(r"^Adjoint\(.*\)$")
     c_adjoint = re.compile(r"^C\(Adjoint\(.*\)\)$")
     adjoint_c = re.compile(r"^Adjoint\(C\(.*\)\)$")
 
-    def is_not_adj(op):
-        return not (re.match(adjoint, op) or re.match(c_adjoint, op) or re.match(adjoint_c, op))
-
-    operations_no_adj = filter(is_not_adj, operations)
-    return set(operations_no_adj)
-
-
-def check_no_overlap(*args, device_name):
-    """Check items in *args are mutually exclusive.
-
-    Args:
-        *args (List[Str]): List of strings.
-        device_name (str): Device name for error reporting.
-
-    Raises:
-        CompileError
-    """
-    set_of_sets = [set(arg) for arg in args]
-    union = set.union(*set_of_sets)
-    len_of_sets = [len(arg) for arg in args]
-    if sum(len_of_sets) == len(union):
-        return
-
-    overlaps = set()
-    for s in set_of_sets:
-        overlaps.update(s - union)
-        union = union - s
+    def is_not_modifier(op):
+        return not (
```

Comment:
Should we match a single regexpr here?  Something like `\^(C(.*)|Adjoint(.*)$`

Reply:
Sure 👍